### PR TITLE
Experiment: API tests using cypress

### PR DESCRIPTION
**Pull request by andrewleith**
# Summary | Résumé

This PR adds cypress and a small suite of tests to test API:

## Admin
- Login
   - Can login to ADMIN and see accounts page
   - Can see Notify service dashboard
- Qualtrics
   - Can see qualtrics survey button on dashboard page and survey opens
- Accessibility
   - GCA - all pages pass a11y and HTML validation at various viewports
   - App - all pages pass a11y and HTML validation 

## API
- Sending email notifications with live, test and team keys:
  - can send email notification without personalisation
  - can send email notification with personalisation
  - can send email to smoke test addresses
  - can use a non-default replyTo
  - can use a default replyTo
  - can use no replyTo   
- Sending file attachments via email notifications:
   - can send single attachment
   - can send 10 attachments
   - cannot send 16 attachments 

## Smoke tests
- API
   - Email: can send/receive a one-off email
   - Email: can send/receive bulk CSV emails
   - Email: can send/receive a one-off email w/ attachment
   - Email: can send/receive one-off emails w/ link attachment
   - SMS: can send a one-off SMS
   - SMS: can send bulk CSV SMSs 
- ADMIN
   - can send/receive a one-off email

## ADMIN Login
I also wrote a test suite to login in to admin and test that the qualtrics survey loads. The tough part of this was being able to do a 2FA login.  A special account was created for this and is used throughout the tests when logging into ADMIN.  This technique works by using a cypress plugin, which allows us to run nodejs code (vs browser js) to poll the inbox using imap.

## Running the tests
- You'll need to add the file `cypress.env.json` into the `/tests_cypress` folder.  It stores the admin auth secret, the admin password, as well as API keys. Its contents can be found in lastpass.
- You can run the smoke tests directly from the dev container using the following command: `npm run smoke`
- Running the tests in the devcontainer doesn't work quite yet, so you'll need to open the folder regularly then start a terminal:
```
% cd tests_cypress 
% npx cypress open
```
- Next, choose **E2E Testing** from the cypress UI, then click **Start E2E testing**.

### CODE REVIEW

File: tests_cypress/config.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+const config = {
+    STAGING: {
+        API: {
+            HostName: "https://api.staging.notification.cdssandbox.xyz",
+        },
+        Admin: {
+            HostName: "https://qvgi7ansvfzeekqiuv6hltlzza0sotpj.lambda-url.ca-central-1.on.aws/", //"https://staging.notification.cdssandbox.xyz",
+            AdminUser: "eli10@ethereal.email",
```
Comment (jimleroyer) [inline tests_cypress/config.js:8]: Brings a good insight: we should put domain validation on login (on top of existing registration validation).

**Comment by jzbahrai** [comment]
how comfortable are you merging in this PR? it looks good to me and there isnt changes to existing functionality in any way

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -24,12 +24,25 @@ RUN apt-get update \
...
+    libgtk-3-0 \
+    libgbm-dev \
+    libnotify-dev \
+    libgconf-2-4 \
+    libnss3 \
+    libxss1 \
+    libasound2 \
+    libxtst6 \
+    xauth \
+    xvfb \
```
Comment (andrewleith) [inline .devcontainer/Dockerfile:45]: These are the necessary dependencies for Cypress to run in the dev container.

**Review by smcmurtry — APPROVED** [review]
Looks great! I checked this out locally, rebuilt my dev container, and was able to run the smoke tests successfully.

Based on the code/files changes, this will have no impact on our app in production, or our CI pipeline, just our local dev environment. So I think this is very low risk to merge and fix any issues as we find them.

After running, I did have git telling my about the generated video file in `cypress/videos/` - let's add that to the gitignore at some point.

**Comment by andrewleith** [comment]
Thanks @smcmurtry!

 I'll gitignore those now so they dont annoy us later!

**Pull request merged**